### ISSUE TITLE
When Steam Deck connect monitor,  touch rotated 90 in game mode.

### ISSUE THREAD

**Issue by kimdonga99** [ValveSoftware/SteamOS#1742]
#### Your system information

* Steam client version:  1733265492
* SteamOS version:  3.6.20
* Opted into Steam client beta?: [No] 
* Opted into SteamOS beta?: [No] 
* Have you checked for updates in Settings > System?: [Yes]

#### Please describe your issue in as much detail as possible:
Describe what you _expected_ should happen and what _did_ happen.

When Steam Deck connect monitor,  touch rotated 90 in game mode.
I've tried all the firmware for almost a year and it still hasn't been fixed.
The only way is to use a version of 3.4 or less.
Considering there was no problem before, it's not a difficult problem.
There are a lot of threads here too, but they still haven't been fixed.

#### Steps for reproducing this issue:

1.  Steam deck Gamemode (big picture)
2.  Connecting to an external monitor
3.  All touch rotated 90

**Comment by denilsonsa** [comment]
I can confirm this bug. Let me give some clear and detailed reproduction steps:

My system:
* Steam Deck original LCD version.
* Version: always up-to-date. Whatever stable version we had in the final weeks of January
* An external USB-C monitor with touch capability.

Example monitors you may want to try:
* [Lenovo ThinkVision M14t Gen 2](https://psref.lenovo.com/Product/ThinkVision/ThinkVision_M14t_Gen_2) - This is the one I have. It supports both touch and an active stylus pen (Lenovo Digital Pen 2 or similar, sold separately).
* [Leovo ThinkVision M14t (Gen 1)](https://psref.lenovo.com/Product/ThinkVision/ThinkVision_M14t) - Older model with slightly lower resolution. It supports both touch and an active stylus pen (included).
* [Verbatim PMT-14](https://www.verbatim-europe.com/en/portable-monitors/products/portable-touchscreen-monitor-14-full-hd-1080p-pmt-14-49591) - It supports touch. Also available in [15"](https://www.verbatim-europe.com/en/portable-monitors/products/portable-touchscreen-monitor-156-full-hd-1080p-pmt-15-49592) and [17"](https://www.verbatim-europe.com/en/portable-monitors/products/portable-touchscreen-monitor-173-full-hd-1080p-pmt-17-49593) models.
* [Wacom One 13 touch (2023)](https://estore.wacom.com/en-us/wacom-one-13-touch-dth134w0a.html) - The newer 2023 model uses a single USB-C cable (while the older gen 1 model used a complicated X-shaped cable). The 13" model has touch support.
* Some *Wacom Cintiq Pro* models may have touch support. They are an order of magnitude more expensive than the others from this list.

## How to reproduce this bug

1. Acquire one USB-C monitor with touch support. It can be one of the ones listed above, or you could find another model. You may also ask a graphic designer friend, they may already have one such display.
2. Switch your Steam Deck to *Desktop Mode*.
3. Plug your monitor to the Steam Deck.
4. Go to *Display Settings* and disable the internal Steam Deck display. Touch coordinates get wonky when there are multiple enabled displays. Thus, don't use *Extend Desktop*, but instead switch to have the desktop only on the external display.
5. Observe how everything just works. The display works, the [multi-finger touch](https://patrickhlauke.github.io/touch/tracker/multi-touch-tracker-pointer-hud.html) works. It's a great experience!
6. Return to *Gaming Mode*.
7. If needed, re-plug your external monitor.
8. If needed, go to Steam Big/Small Picture settings and adjust the display settings for the correct resolution.
9. Launch any game, or a browser, or anything that requires interacting with a (mouse) pointer. Anything that makes it clear where the touch is being detected.
10. BUG! Observe how the touch coordinates are now rotated 90°.

In summary:

| | Desktop mode | Gaming mode |
|-|-|-|
| Built-in Steam Deck display | Touch works | Touch works |
| External USB-C display      | Touch works | Touch is rotated 90° |

## Untested scenarios

* Active pressure-sensitive stylus pen while in Gaming mode.
* Rotated displays. Some displays have automatic orientation detection, and they automatically communicate it to the OS. (Sidenote: I wish I could rotate the display in the Gaming mode settings.)

**Comment by denilsonsa** [comment]
I recorded some videos that clearly show the issue. And sorry for the [Vertical Video Syndrome](https://www.youtube.com/watch?v=f2picMQC-9E), but that was the best way to make sure both screens were in-frame.

Tested using: <https://patrickhlauke.github.io/touch/tracker/multi-touch-tracker-pointer-hud.html>

Tested on Google Chrome [because it has the best touch+pen support with fewer bugs](https://stackoverflow.com/questions/78073830/pen-pointer-events-in-linux-chrome-and-firefox-not-working-as-intended/78764151#78764151)

## Gaming mode

Not shown: active stylus pen is not supported in gaming mode.

https://github.com/user-attachments/assets/e7189e26-c635-4b60-a100-b336ddce8eb9

## Desktop mode

Everything works correctly in Desktop mode.

https://github.com/user-attachments/assets/717f97c1-ecd9-4fed-b2bc-bdd7e38b3e2f

At the end of the video, because the screen was extended (two displays), the touch coordinates were incorrect. This can easily be fixed by disabling the built-in screen. Alternatively, if SteamOS shipped with KDE's tablet KCM (KDE Configuration Manager) plugin, we could configure it over there.
